library/src/main/scala/giter8/launcher/Launcher.scala

```diff
@@ -0,0 +1,132 @@
+package giter8.launcher
+
+import java.io.{File, FileInputStream}
+import java.util.Properties
+
+import giter8.{TempDir, _}
+import org.apache.commons.io.FileUtils
+
+import scala.io.Source
+import scala.util.{Failure, Success, Try}
+
+object Launcher extends TempDir {
+
+  import giter8.G8.RichFile
+  import giter8.OptTry
+
+  private val git = new Git(new JGitInteractor())
+
+  /**
+    * Use JAVA_HOME as a default location for finding a java executable. If
+    * it isn't set fall back to the `java.home` of the current JVM
+    */
+  private lazy val javaHome: String =
+    sys.env.getOrElse("JAVA_HOME", System.getProperty("java.home"))
+
+  /**
+    * Resolves the version of giter8 to be launched.
+    *
+    * If `g8Version` is defined, that will be used. Otherwise, we clone
+    * the template repo and look for a `giter8.version` property in `project/build.properties`
+    *
+    * Returns `None` if we can't find a version from either of these
+    *
+    * @param g8Version
+    * @param template
+    * @return
+    */
+  private def version(g8Version: Option[String], template: String): Try[Option[VersionNumber]] =
```

Comment:
Another fallback it should consider is JVM flat `-Dgiter8.version`, to be consistent with sbt's `sbt.version`.